From: Reinhard Tartler
Subject: Fixes mkv playback with gcc-4.4 as found in debian/squeeze

This effectively reverts this upstream commit:
http://git.videolan.org/?p=vlc/vlc-2.0.git;a=commitdiff;h=59491dcedffbf97612d2c572943b56ee4289dd07

Suggested by:
http://forum.videolan.org/viewtopic.php?f=13&t=98175&p=327945&hilit=KaxSegment#p327945

Thanks to Artyom Kazak <artyom.kazak@gmail.com> for pointing this out

Index: vlc/modules/demux/mkv/mkv.hpp
===================================================================
--- vlc.orig/modules/demux/mkv/mkv.hpp	2012-08-10 13:14:35.284287024 +0200
+++ vlc/modules/demux/mkv/mkv.hpp	2012-08-10 13:15:13.839249260 +0200
@@ -115,7 +115,7 @@
 
 #define MKVD_TIMECODESCALE 1000000
 
-#define MKV_IS_ID( el, C ) ( el != NULL && typeid( *el ) == typeid( C ) )
+#define MKV_IS_ID( el, C ) ( EbmlId( (*el) ) == C::ClassInfos.GlobalId )
 
 
 using namespace LIBMATROSKA_NAMESPACE;
